### DISCUSSION TITLE
Read only and prepopulated fields in custom edit forms

### DISCUSSION THREAD

**Discussion by KellerKev** [emmett-framework/emmett#326] [Q&A] [answered]
Just a bunch of other questions. I come from the py4web / web2py world, but I am very curious about Emmett and I want to give it a serious shot. 

1) When a create a form and I want a read only field, I would in the pydal world set it to field.writable=False or if I want to omit the field I would set it to field.readable=False. 

2) If I would create a form based on a list of fields and if someone wants to use the same form to edit data, I would need to query the id of the table, fetch the values and prepopulated the field values using field.default=value. Is there a field.default in Emmett or how do you do it?

3) Someimes I want to break up form fields in the template for design purposes or in order to customize the form further. 
For example in the template. 

{=form.begin}
{=form.field.label}
{=form.field.}
{=form.submit}
{=form.end}

Can I do this in Emmett as well in the template?

Thank you!

**Comment by gi0baro** [comment, marked as answer]
With the same order:

1. In Emmett the parameter is `rw` (https://emmett.sh/docs/2.2.x/orm/models#fields-accessibility), which accepts a boolean (in case you want to set both readability and writability to `True` or `False`) or a tuple (`(True, False)` means readable but not writable).
2. `Field` has a `default` parameter as well in Emmett (https://emmett.sh/docs/2.2.x/orm/models#default-values). In case you use ORM and models, probably using `Model.form(record)` is easier (https://emmett.sh/docs/2.2.x/forms#forms-with-database-entities)
3. It is not documented, but forms in Emmett have the same `custom` flow of web2py (code ref: https://github.com/emmett-framework/emmett/blob/master/emmett/forms.py#L197#L233), so you can stick with the same code

**Comment by KellerKev** [comment]
Thank you!
